internal/command/action_test.go

```diff
@@ -120,6 +127,121 @@ func TestRunAction(t *testing.T) {
 
 		So(string(content), ShouldEqual, expectedValue)
 	})
+
+	Convey("Docker options correctly injected", t, func() {
```

Comment:
So, from my understanding of how things _currently_ work, we only support the following syntax:
```
summon docker @SUMMONDOCKERARGS someImage someCommand
```

That being said, we should also have tests for the following situations:
```
summon /bin/bash -c "docker @SUMMONDOCKERARGS someImage someCommand"
```
```
summon --yaml "VAR: !var some/path" /bin/bash -ec "
docker @SUMMONDOCKERARGS someImage someCommand
echo $VAR
"
```
Although these may not be actual use cases, we should remove as much ambiguity as we can from possible situations.

Reply:
We should also support the case where someone also uses @SUMMONENVFILE and @SUMMONDOCKEROPTS at the same time

Reply:
I think it's supported by default. The 2 approaches aren't coupled in any way, and so shouldn't interact in anyway (at least on the Summon side of things).
1. @SUMMONENVFILE is just a string that points to an env file containing all the resolved values.
2. @SUMMONDOCKEROPTS output the Docker CLI options, which are combinations of `--env` and `--volume`, such that the environment variables injected by summon to the child process can become available inside a container created with these options.

Reply:
That's understandable, but my concern would be changes we might make in the future surrounding how we parse these keywords prefixed with '@'. I personally think it's safer to have tests in place now that ensure that changes don't have unintended conflicts in the future, and the features have enough overlap that it may be good to have a test, just in case. What do you think @diverdane ?

Reply:
I thought about this a bit. I think a good way to proceed is to 
1. Acknowledge that the 2 approaches are actually entirely unrelated, particularly in terms of implementation so their implementations need not be tested in unison.
2. Since they both result in modification of the arguments of the child process we can have a simple test that ensures that replacing doesn't get messed up.